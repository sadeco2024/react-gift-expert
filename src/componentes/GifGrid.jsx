

import { GifGridItem } from "./GifGridItem";
import { useFetchGifts } from "../hooks/useFetchGifts";


export const GifGrid = ({categoria}) => {
    
  const {images, isLoading} = useFetchGifts(categoria) ;

  return (
    <>
        <h4>{categoria}</h4>
        {
          isLoading && (<h2>Cargando...</h2>)
        }
        <div className="card-grid">
          { 
            images.map(( image) => (
             <GifGridItem
              key={image.id}
              { ...image}
             />
          ))}
          
        </div>
        {/* <h5>{counter}</h5>
        <button onClick={() => setCounter(counter + +1) }>+1</button> */}
    </>
  )
}
