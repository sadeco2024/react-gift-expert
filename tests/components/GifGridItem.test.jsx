import { render,screen } from "@testing-library/react";
import { GifGridItem } from "../../src/componentes/GifGridItem";


describe('Pruebas <GifGridItem/>', () => { 
      
    const url = 'http://sadeco.mx/';
    const title='PÃ¡gina de Sadeco';
 
    test('debe hacer match con el snapshot', () => { 

        const {container} = render( <GifGridItem title={title} url={url}  />);
        expect(container).toMatchSnapshot();

       })

       test('Debe mostrar la imagen con el URL y el ALT indicado.', () => { 
            const {container} = render( <GifGridItem title={title} url={url}  />);
            // screen.debug();

            // expect(screen.getByRole('img').src).toBe(url);
            // expect(screen.getByRole('img').alt).toBe(title);

            const {src,alt} = screen.getByRole('img');
            expect(src).toBe(url);
            expect(alt).toBe(title);

        })

 })