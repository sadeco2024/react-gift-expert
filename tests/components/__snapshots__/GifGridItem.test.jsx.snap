// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <GifGridItem/> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Página de Sadeco"
      src="http://sadeco.mx/"
    />
    <p>
      Página de Sadeco
    </p>
  </div>
</div>
`;
